Sepia extract rebalance [TM Cand/Ready]

4dplanner opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

reworks sepia extract and some crossbreeds for new effects.

todo: 

- [x] mechs are vulnerable to sepia

- [x] robots are vulnerable to sepia 

- [x] antimagic only works against magic timestop
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

Point by point:

Core change: sepia slimes are very overpowered currently, being a long stun that ignores stunimmunity, blockchance, doesn't require aiming, and covers a 5x5 area. I think having a delay on them makes them an interesting area denial tool/guaranteed kill if you have a gun.

Recurring sepia: This was insanely powerful. You could run around spamming timefields like there was no tomorrow. However, I didn't want to just make it useless - I think the recalling effect will make sure it fills the intended role of a convenient and sustainable extract. In fact this crossbreed was so overpowered it overshadowed all of the other sepia crossbreeds, meaning they barely got used.

Chilling sepia: As above, but it'll be interesting to see how the teamwork dynamic works out. I think it'll open up some interesting strategies, while being bad for a solo murderboner player.

Burning sepia: The timefreeze camera was as mentioned before overshadowed by the recurring sepia, but is still very powerful. Combined with a laser you can guarantee a kill with a single undodgeable instant travel click at long range, which is crazy.

## Changelog
:cl:
add: sepia slime extract has a delay before activating
balance: the user of a sepia slime extract is no longer immune to the time field
balance: chilling sepia is now the support extract - it always freezes the user, but not other marked people
balance: give one to your golem!
add: burning sepia spawns the rewind camera. Take a selfie with someone and give it to them to make sure they remember the moment forever!
balance: you don't actually need to give them the photo, they'll remember anyway. Keep it as a reminder.
add: rewinding resets your location to the turf you were on after 10 seconds, and heals simple animals and objects.
remove: timefreeze camera is admin spawn only.
add: recurring sepia recalls to the hand of the last person to touch it after activating. Reusable timestop grenades!
Tweak: some special cameras don't prompt for customisation
fix: timefreeze camera actually makes a photo 
fix: timestop stops pathing 
fix: timestop stops mechs
fix: adds a check to make sure simple_animals don't get their AI toggled while sentient
fix: Adds the timestop overlay to frozen projectiles
fix: Timestopped things have INFINITY move_resist as opposed to being anchored
fix: Timestop will now unfreeze things that somehow leave it
fix: mobs in the middle of a walk_to will have their walk stopped by timestop
fix: mobs that are stunned will be stopped mid walk as well
fix: Slimes respect mobility_flags & MOBILITY_MOVE
fix: Slimes no longer automatically regain MOBILITY_MOVE whenever not cold
fix: pulling respects changes in move_force
fix: swapping places respects move_force if the participant is not willing
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

**Cobby's Note:** I misclicked okay don't worry :)

4dplanner committed:
Sepia slime has a timer before activating

4dplanner committed:
Merge branch 'master' into sepia

4dplanner committed:
Chilling and recurring

4dplanner committed:
Rewind camera

4dplanner committed:
Camera variable to disable customisation

4dplanner committed:
Removes unnecessary selfieing

4dplanner committed:
Changes rewind time to 10 seconds

Repukan commented:
You know what this PR needs?

Markass Brawnlee

4dplanner commented:
that's hot

imsxz commented:
how does the rewind camera work? seems really neat from a quick glance at the code but im a brainlet and having trouble grasping it entirely.

4dplanner commented:
> how does the rewind camera work? seems really neat from a quick glance at the code but im a brainlet and having trouble grasping it entirely.

You click the guy, he goes around as normal, and ten seconds later he jumps back to the turf he was on when it started.

4dplanner commented:
Bonus: if they are a simplemob/an object, they heal up as well.

zxaber commented:
>mechs are vulnerable to sepia

Farewell to my dabbing on admemes that try to trap me with a timestop, only for me to stomp away freely.

4dplanner committed:
Simple animal fixes

4dplanner committed:
Merge remote-tracking branch 'upstream/master' into sepia

4dplanner committed:
properly freezes bots

ExcessiveUseOfCobblestone commented:
Antimagic not working on science timestops will be okay, I'd rather balance with that then "JUST GET A MELON IF YOU HATE XENOBIO" arguments.

4dplanner commented:
Looking at the code, they should actually already be immune. Bug maybe?

4dplanner commented:
In fact testing seems to suggest that antimagic doesn't work on slime timestop at all, and I'm not really sure why people keep suggesting it as a counter

ExcessiveUseOfCobblestone commented:
no one (including me) reads code lol

4dplanner committed:
Mech timestop

4dplanner committed:
Merge remote-tracking branch 'upstream/master' into sepia2

4dplanner committed:
Merge branch 'sepia' into sepia2

4dplanner committed:
P -> M

4dplanner committed:
Gives a warning message

knacker48 commented:
People keep talking like xenobiologist is the most overpowered job yet I swear they always die first since they're so Isolated which makes them an easy kill. Also all of the good stuff usually comes after the threat has passed or everyone's dead and the shuttle left

Nickkoranda commented:
I haven't really ever seen anybody go out of their way to kill xenobio specifically unless they have a huge hate boner for it (lexia black). Generally, the isolation is what makes it useful since people would rather roam around the central hallway for victims than to break into departments.

4dplanner committed:
This should have commited with the last set

ExcessiveUseOfCobblestone commented:
yeah xenobio is pretty underdeveloped with admin revives, instant death bolts, ability to create an army of golems or simplemobs each with their own unique abilities, ability to start fires, ability to bomb, ability to convert people into borgs, and whatever the latest and greatest wacky crossbreeding is.

ExcessiveUseOfCobblestone commented on code/modules/reagents/chemistry/recipes/slime_extracts.dm:
```diff
@@ -559,9 +559,17 @@
+	if(istype(extract))
+		if(extract.Uses > 0)
+			var/mob/lastheld = get_mob_by_key(holder.my_atom.fingerprintslast)
+			if(lastheld)
```
```suggestion
			if(lastheld && !lastheld.equip_to_slot_if_possible(extract, SLOT_HANDS, disable_warning = TRUE))
```

ExcessiveUseOfCobblestone replied:
this suggestion is bait since you need to fix 570/1

4dplanner committed:
Just screw my line lengths up

ExcessiveUseOfCobblestone commented:
looks ok but cringing at the fact there's like 5 lists that may or may not be used everytime timestop is made

4dplanner commented:
Should I compress them down to two lists?

4dplanner commented:
Actually probably don't need the throwing list either, will jumble it all in

ExcessiveUseOfCobblestone commented:
awesome just ping me when it's done and ill merge

optimumtact commented:
it's more correct to say the rewards are developed, but the actual gameplay of the job is extremely lacking, because slimes are super easy to care for.

Shadowflame909 commented:
@ExcessiveUseOfCobblestone None of those can beat 1 memer boy hunting their target in a place where no one will bother them before an hour in.

And once an hour in hits, all of those can't beat a gygax or a durand.

Angustmeta commented:
Fixes #37249 , an old issue I found while browsing.

knacker48 commented:
@Angustmeta That's how I see it too. There's a lot of randomness when it comes to how the slimes split. For instance if you are trying to rush rainbow slimes you just have to hope that you get lucky and get a orange and that orange splits into a red. I've had a couple of rounds where I couldn't get a red until the 40 minute mark, at which point I basically gave up on it because the shuttle was called

optimumtact commented:
that only illustrates the point though.

The only complexities are attention and RNG, there's limited actual deep engagment required.

ExcessiveUseOfCobblestone commented:
> @ExcessiveUseOfCobblestone None of those can beat 1 memer boy hunting their target in a place where no one will bother them before an hour in.
> 
> And once an hour in hits, all of those can't beat a gygax or a durand.

The only downside is the double-edged sword of being in fort knox and that you can't kill a single, specific item instantly (combat mechs)? lol

4dplanner committed:
Big changes

Adds pre-move COMSIG
Cuts down on lists in timestop
Adds the overlay to frozen projectiles
Timestopped things have INFINITY move_resist as opposed to being
anchored
Timestop will now unfreeze things that somehow leave it
Cleans up timestop code
mobs in the middle of a walk_to will have their walk stopped by timestop
mobs that are stunned will be stopped mid walk as well
Slimes respect mobility_flags & MOBILITY_MOVE
Slimes no longer automatically regain MOBILITY_MOVE whenever not cold

4dplanner commented:
Needs new review, I changed it up a lot

4dplanner commented:
Changelog grows longer

ExcessiveUseOfCobblestone commented:
die die die die die

optimumtact commented:
u fricken crab, remove those code: ones, that's what the git logs are for

4dplanner committed:
Pulling and swapping places respect move_force

4dplanner commented:
Going to need a testmerge, messing with pulling and stuff

ExcessiveUseOfCobblestone commented on code/modules/research/xenobiology/crossbreeding/_misc.dm:
```diff
@@ -4,27 +4,93 @@ Slimecrossing Items
+	else if(istype(parent, /obj))
+		var/obj/O = parent
+		O.obj_integrity = health
+	rewinds_remaining -= 1
```
```suggestion
	rewinds_remaining --
```

ExcessiveUseOfCobblestone replied:
@4dplanner

4dplanner replied:
oops

ExcessiveUseOfCobblestone commented on code/modules/research/xenobiology/crossbreeding/_misc.dm:
```diff
@@ -4,27 +4,93 @@ Slimecrossing Items
+		return
+	if(!used)//selfie time
+		if(user == target)
+			to_chat(user, "<span class=notice>You take a selfie!</span>")
```
cringed

optimumtact replied:
selfie time

ExcessiveUseOfCobblestone committed:
Update code/modules/research/xenobiology/crossbreeding/_misc.dm

Co-Authored-By: 4dplanner <3combined@gmail.com>

Angustmeta commented:
Tested these today and they seem fine. 

But timefreeze makes you drop items. Can't remember if it was always like that.

4dplanner commented:
> Tested these today and they seem fine.
> 
> But timefreeze makes you drop items. Can't remember if it was always like that.

It's always had the item drop aspect, yes. Thanks for the testing!
